### PR TITLE
CBL-4506: Replicator starts up slow for big database

### PR DESCRIPTION
**Pull request by jianminzhao**
The underlying cause has been fixed in #711f886899b23b. We added a test case.

### CODE REVIEW

File: C/tests/c4DatabaseTest.cc [hunk 1]
```diff
@@ -1011,6 +1014,69 @@ static void testOpeningOlderDBFixture(const string& dbPath, C4DatabaseFlags with
...
+    double elapsedUnsorted = doIt(false);
+    double elapsedSorted   = doIt(true);
+    cout << "Enum with conflicted/sorted takes " << elapsedSorted << " sec, Enum With conflicted/unsorted takes "
+         << elapsedUnsorted << "sec, ratio = " << elapsedSorted / elapsedUnsorted << endl;
+    auto config = c4db_getConfig2(db);
+    if ( config->encryptionKey.algorithm != kC4EncryptionNone ) {
+        // Encrypted case is not stable.
+        return;
+    }
+    CHECK(elapsedSorted / elapsedUnsorted > 5);
```
Comment (pasin) [inline C/tests/c4DatabaseTest.cc:1077]: Could this be flaky when running on some machines?

Comment (jianminzhao) [reply C/tests/c4DatabaseTest.cc:1077]: I picked number, 5, quite conservatively. On my machine, the ratio is always > 10, but I found over the weekend that jenkins' windows build may be smaller. The theory is, the more docs the db has, the higher the ratio, I don't want to put too much time on this test. Perhaps this test does not belong here, but I wanted to set a guard here lest in the future we somehow we make changes to cause the problem reappear.

Comment (borrrden) [reply C/tests/c4DatabaseTest.cc:1077]: It sounds like a performance test, perhaps it should be run as part of the perf suite for showfast?

**Review by borrrden — APPROVED** [review]

File: C/tests/c4DatabaseTest.cc (hunk 1 above)
Comment (jianminzhao) [reply C/tests/c4DatabaseTest.cc:1077]: Definitely. I should get familiar with that.

**Pull request merged**